test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -396,6 +396,53 @@
             #"shutdown-on-error triggered because of exception"
             :error))))))
 
+(deftest ^:integration debouncing-test
+  (testing "Default debouncing behavior"
+    (let [root-dir (fs/temp-dir "debounce")
+          actual-events (atom [])
+          actual-calls (atom 0)
+          callback (fn [e]
+                     (swap! actual-events concat e)
+                     (swap! actual-calls inc))
+          app (tk/boot-services-with-config watch-service-and-deps {})
+          svc (tk-app/get-service app :FilesystemWatchService)
+          event-cadence (quot watch-core/window-min 2)]
+      ;; We trigger half the number of events that could independently occur
+      ;; without exceeding the window-max, we trigger them at half the wait
+      ;; of window-min to ensure they will all be caught even if the host
+      ;; running the tests is slow. This requires that window-max be at least
+      ;; twice the duration as window-min.
```

Comment:
I think race conditions are going to be inevitable in any non-trivial test cases like this.  I think this comment should just be updated to clarify the wording - specifically "ensure" is a little misleading, I think.